modules/azure/aks.go

```diff
@@ -2,29 +2,35 @@ package azure
 
 import (
 	"context"
+	"fmt"
 
 	"github.com/Azure/azure-sdk-for-go/services/containerservice/mgmt/2019-11-01/containerservice"
 	"github.com/gruntwork-io/terratest/modules/testing"
 )
 
 // GetManagedClustersClientE is a helper function that will setup an Azure ManagedClusters client on your behalf
 func GetManagedClustersClientE(subscriptionID string) (*containerservice.ManagedClustersClient, error) {
-	// Validate Azure subscription ID
-	subscriptionID, err := getTargetAzureSubscription(subscriptionID)
+	// Create a cluster client
+	factory := NewClientFactory()
+	client, err := factory.GetClientE(ManagedClustersClientType, subscriptionID)
```

Comment:
To help my understanding: should _all_ Azure/Terratest code using this factory from now on? I know we're not updating it all in this particular PR, but is the expectation that we do update it this way in the future?

If that's the case, and apologies if this is a dumb question, but should this factory not be part of the AWS SDK itself then?

Reply:
The Azure SDK does in fact have support for leveraging multi-environment clouds. However, it require exactly the steps this code is hiding, and is something we wanted to keep away from the concern of module writers. With that said, we do intend to deliver this feedback to the SDK team to help make this easier.

Reply:
Yes - all future modules should be using this factory (in subsequent PR's)